plugins/fabrik_element/fileupload/fileupload.php

```diff
@@ -457,11 +458,6 @@ protected function ajaxFileFilters()
 		$return = new stdClass;
 		$exts = $this->_getAllowedExtension();
 
-		foreach ($exts as &$ext)
```

Comment:
Yes - the pregreplace change later removes the leading . on an extension, so this loop is not needed.

Though I think there is a corner case (. as first character) which needs to be added to the pregreplace